### PR TITLE
boards: shields: Add X-NUCLEO-IDB05A1

### PR DESCRIPTION
**Pull request by yboutreux**
Add STMicroelectronics BLE shield X-NUCLEO-IDB05A1

Signed-off-by: Yaël Boutreux <yael.boutreux@st.com>

### CODE REVIEW

File: boards/shields/x_nucleo_idb05a1/x_nucleo_idb05a1.overlay [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/*
+ *
+ * Copyright (c) 2019 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&arduino_spi {
+
```
Comment (arnop2) [inline boards/shields/x_nucleo_idb05a1/x_nucleo_idb05a1.overlay]: Suppress blank line

File: boards/arm/stm32mp157c_dk2/arduino_r3_connector.dtsi [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+			   <18 0 &gpioe 13 0>,	/* D12 */
+			   <19 0 &gpioe 12 0>,	/* D13 */
+			   <20 0 &gpioa 12 0>,	/* D14 */
+			   <21 0 &gpioa 11 0>;	/* D15 */
+	};
+};
+
+/*
+ * arduino_i2c is located on ANA0 and ANA1,
+ * so there can't be an arduino_i2c definition
```
[the remark this answers is not in the document]
Comment (yboutreux) [reply boards/arm/stm32mp157c_dk2/arduino_r3_connector.dtsi]: Didn't saw that Arduino has also I2C on D14 and D15, my bad !

**Comment by yboutreux** [comment]
> Great!
> Split the shield and the board commit, so we don't have to wait for dependency.
> Then go for zephyr upstream.
> 
> Question: Have you been able to test the shield with any board other than MP1?

Not yet, I only checked with the MP1, tell me if you want me to check with something else and I'll try some basic samples BLE samples :)

File: boards/arm/stm32mp157c_dk2/arduino_r3_connector.dtsi (hunk 2 above)
[the remark this answers is not in the document]
Comment (yboutreux) [reply boards/arm/stm32mp157c_dk2/arduino_r3_connector.dtsi]: This part is moved to another PR

File: boards/shields/x_nucleo_idb05a1/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+Overview
+********
+The X-NUCLEO-IDB05A1 is a Bluetooth Low Energy evaluation board based on the
+SPBTLE-RF BlueNRG-MS RF module to allow expansion of the STM32 Nucleo boards.
+The SPBTLE-RF module is FCC (FCC ID: S9NSPBTLERF) and IC certified
+(IC: 8976C-SPBTLERF).
+
+The X-NUCLEO-IDB05A1 is compatible with the ST Morpho and Arduino UNO R3
+connector layout (the user can mount the ST Morpho connectors, if required). The
+X-NUCLEO-IDB05A1 interfaces with the STM32 microcontroller via the SPI pin, and
```
[the remark this answers is not in the document]
Comment (yboutreux) [reply boards/shields/x_nucleo_idb05a1/doc/index.rst]: Updated, replaced "STM32" to "host"

File: boards/shields/x_nucleo_idb05a1/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+More information about the board can be found at the
+`X-NUCLEO-IDB05A1 website`_.
+
+Hardware
+********
+
+X-NUCLEO-IDB05A1 provides a SPBTLE-RF chip with the following key features:
+
+
```
[the remark this answers is not in the document]
Comment (yboutreux) [reply boards/shields/x_nucleo_idb05a1/doc/index.rst]: Extra-lines removed

**Comment by yboutreux** [comment]
@erwango what about this :
IDB05A1 PR : no dependencies, ready to upstream
R3 connector PR : no dependencies, ready to upstream
SPI PR : dependent of R3 connector, wait for R3 to be added, then add the SPI node to the Arduino R3 mapping

**Comment by yboutreux** [comment]
Updated with more informations: specified which test passed on which boards.
Added disco_l475_iot1.conf as a dummy to document BlueNRG-MS chip 16MHz know limitation

**Comment by yboutreux** [comment]
Waiting status about https://github.com/zephyrproject-rtos/zephyr/issues/17310 to go further

File: boards/shields/x_nucleo_idb05a1/boards/disco_l475_iot1.conf [hunk 5]
```diff
@@ -0,0 +1 @@
+CONFIG_SPI_1=y
```
Comment (erwango) [inline boards/shields/x_nucleo_idb05a1/boards/disco_l475_iot1.conf]: Well, not sure it makes sense to enabled this shield on this board.

File: boards/shields/x_nucleo_idb05a1/boards/stm32mp157c_dk2.conf [hunk 6]
```diff
@@ -0,0 +1 @@
+CONFIG_SPI_4=y
```
Comment (erwango) [inline boards/shields/x_nucleo_idb05a1/boards/stm32mp157c_dk2.conf]: If SPI_4 is arduino SPI, I'd expect it to be enabled conditionally if SPI is enabled

File: boards/shields/x_nucleo_idb05a1/boards/disco_l475_iot1.conf (hunk 5 above)
Comment (yboutreux) [reply boards/shields/x_nucleo_idb05a1/boards/disco_l475_iot1.conf]: Well, indeed this is twisted... Will be removed

File: boards/shields/x_nucleo_idb05a1/boards/stm32mp157c_dk2.conf (hunk 6 above)
Comment (yboutreux) [reply boards/shields/x_nucleo_idb05a1/boards/stm32mp157c_dk2.conf]: correct this is Arduino SPI, will be removed

**Comment by yboutreux** [comment]
@erwango changes dones

**Comment by erwango** [comment]
@yboutreux, ok to upstream after you fix the conflict

**Comment by yboutreux** [comment]
Closed because went to upstream

**Pull request closed without merging**